packages/nimble/R/registration.R

```diff
@@ -17,6 +17,8 @@ if(!exists('.NimbleUseRegistration') || !.NimbleUseRegistration) {
   ## R_setSinglePtrFromSinglePtr = 'setSinglePtrFromSinglePtr'
    C_dwish_chol = 'C_dwish_chol'
    C_rwish_chol = 'C_rwish_chol'
+   C_dinvwish_chol = 'C_dinvwish_chol'
```

Comment:
C_dinvwish_chol is needed as the entry point for R functions to make use of dinvwish_chol. In particular, the dinvwish_chol function in R calls C_dinvwish_chol, which then dispatches to the core C++ function for dinvwish_chol. 

So I don't think you need a C_nimOptim, since R's optim will itself call the needed C++ code.